src/etc/cat-and-grep.sh

```diff
@@ -51,6 +51,15 @@ while getopts ':vieh' OPTION; do
     esac
 done
 
+# an utility function to check if a string contains a substring
+stringContain() { [ -z "$1" ] || { [ -z "${2##*$1*}" ] && [ -n "$2" ];};}
+
+if ! stringContain 'E' "$GREPFLAGS"
```

Comment:
`echo "$GREPFLAGS" | grep -q E` seems simpler here?

Reply:
Yes. Using grep is more simple (at the cost of calling an external program).